tests/recorded/issue75.txt

```diff
@@ -1,3 +1,12 @@
+# The original issue was about shed requiring two passes to format this:
+# on first run autoflake would remove imports, and second run isort would collapse
+# to a single line of imports.
+# Now we have slightly different behaviour: black will add a trailing comma to the
+# import list, which ruff then will respect when removing imports, such that it doesn't
+# collapse the imports to a single line.
```

Comment:
suggests that we might want to run `ruff` before `black` then?

Reply:
This messed up pyupgrade ... which made me realize that ruff has pyupgrade rules as well.

Ruff needs to run after codemods (to remove needless pass inserted by `split_assert`), but I don't see any reason why black needs to run before codemods (and running it after fixes one small case in `split_assert`), so changed it to do codemods->ruff->black.

Reply:
```py
(
    []
    .append(1)
)
```
when `pointless_parens` tries to remove parens here it doesn't delve into the `Call` node to remove the newline, so removing the parens fail. (and trying to do so sounds like a Bad Idea:tm:) But black unlocks this fix. So a naive codemod->ruff->black will change stuff on second run.

Another case that depends on ordering, which would get improved by not running black first, is
```py
# input
assert (
a # a
and b # b
)
# black -> codemod (status quo)
assert a  # a  # b
assert b
# codemod -> black
assert a  # a
assert b  # b
```

I don't feel like it matters a ton either way, so I suggest keeping status quo of black->codemods->ruff->[black]